src/NUnitFramework/framework/Constraints/ExactCountConstraintResult.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections;
+using System.Text;
+
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>
+    /// Contain the result of matching a <see cref="ExactCountConstraint"/> against an actual value.
+    /// </summary>
+    public sealed class ExactCountConstraintResult : ConstraintResult
+    {
+        /// <summary>
+        /// The actual value to which the Constraint was applied.
+        /// </summary>
+        private readonly object _actualValue;
+
+        /// <summary>
+        /// Constructs a <see cref="ConstraintResult"/> for a <see cref="ExactCountConstraint"/>.
+        /// </summary>
+        /// <param name="constraint">The Constraint to which this result applies.</param>
+        /// <param name="actualValue">The actual value to which the Constraint was applied.</param>
+        /// <param name="isSuccess">If true, applies a status of Success to the result, otherwise Failure.</param>
+        public ExactCountConstraintResult(IConstraint constraint, object actualValue, bool isSuccess)
+            : base(constraint, actualValue, isSuccess)
+        {
+            _actualValue = actualValue;
+        }
+
+        /// <summary>
+        /// Write the actual value for a failing constraint test to a MessageWriter.
+        /// </summary>
+        /// <param name="writer">The writer on which the actual value is displayed</param>
+        public override void WriteActualValueTo(MessageWriter writer)
+        {
+            var elementList = _actualValue as IEnumerable;
+            if (elementList == null)
+            {
+                writer.Write("no items");
+                return;
+            }
+
+            var itemCounter = 0;
+
+            foreach (var element in elementList)
+            {
+                itemCounter++;
+            }
+
+            if (itemCounter == 0)
+            {
+                writer.Write("no items");
+                return;
+            }
+
+            var writeCounter  = 0;
+            var stringBuilder = new StringBuilder();
```

Comment:
It's more efficient and simpler code to read if you write directly to the writer (which may already be wrapping a StringBuilder) than to start a buffer here.